backend/src/apiserver/resource/model_converter.go

```diff
@@ -111,6 +111,30 @@ func (r *ResourceManager) ToModelJob(job *api.Job, swf *util.ScheduledWorkflow,
 	}, nil
 }
 
+func (r *ResourceManager) ToModelPipelineVersion(version *api.PipelineVersion) (*model.PipelineVersion, error) {
+	codeSources := make([]byte, 0)
+	if version.CodeSourceUrls != nil && len(version.CodeSourceUrls) > 0 {
+		for _, codeSource := range version.CodeSourceUrls {
+			codeSources = append(codeSources, codeSource...)
+			codeSources = append(codeSources, ';')
```

Comment:
store as json format instead to avoid manual processing which is error prone
```
    var urls = []string{
        "http://google.com",
        "http://facebook.com",
        "http://youtube.com",
        "http://yahoo.com",
        "http://twitter.com",
        "http://live.com",
    }

    urlsJson, _ := json.Marshal(urls)
```

Reply:
done